### PR TITLE
Throw error on setup failure, add new branch name management

### PR DESCRIPTION
**Pull request by plocket**
Working for https://github.com/GBLS/docassemble-MAEvictionDefense currently

### CODE REVIEW

File: lib/utils/env_vars.js [hunk 1]
```diff
@@ -13,11 +13,16 @@ const env_vars = {
...
   PLAYGROUND_PASSWORD: process.env.PLAYGROUND_PASSWORD,
   BASE_URL: process.env.BASE_URL,
-  BRANCH_NAME: process.env.BRANCH_PATH && process.env.BRANCH_PATH.split('/')[1] || 'test',
   REPO_URL: process.env.REPO_URL,
   DEBUG: process.env.DEBUG,
 };
 
+// Path could be 'refs/branch_name' or 'refs/head/branch_name' and maybe other things
+// Assume branch name is last part of branch path
+let split = process.env.BRANCH_PATH.split('/');
```
Comment (niharikasingh) [inline lib/utils/env_vars.js:22]: Maybe this should be in the if block following. In my Chrome console, the following lead to errors: 

```javascript
x = null 
x.split('/')
// ERROR 
x = undefined 
x.split('/')
// ERROR 
```

**Review by niharikasingh — CHANGES_REQUESTED** [review]

File: lib/github_integration/install-branch.js [hunk 2]
```diff
@@ -10,7 +10,8 @@ const setup = async () => {
     await waitForPage(page);  // Ensure server has really restarted
   }
   catch (e) {
-    console.log(e);
+    await page.screenshot({ path: `error-setingUp.jpg`, type: 'jpeg', fullPage: true });
```
[the remark this answers is not in the document]
Comment (plocket) [reply lib/github_integration/install-branch.js:13]: Good catch 👍

File: lib/utils/env_vars.js (hunk 1 above)
Comment (plocket) [reply lib/utils/env_vars.js:22]: Makes sense. I was thinking we'd want it to error if the branch path was messed up, but we're providing a fallback already. There are probably pros and cons about _that_ to talk about too, but I'm not sure how crucial that conversation is, so we might as well stick with what we're doing now.

Comment (plocket) [reply lib/utils/env_vars.js:22]: I suppose we then don't need the check on line 23 to check for the existence of the item on the list as even a single item will be valid.

File: lib/utils/env_vars.js [hunk 3]
```diff
@@ -13,11 +13,19 @@ const env_vars = {
...
   PLAYGROUND_PASSWORD: process.env.PLAYGROUND_PASSWORD,
   BASE_URL: process.env.BASE_URL,
-  BRANCH_NAME: process.env.BRANCH_PATH && process.env.BRANCH_PATH.split('/')[1] || 'test',
   REPO_URL: process.env.REPO_URL,
   DEBUG: process.env.DEBUG,
 };
 
+// Path could be 'refs/branch_name' or 'refs/head/branch_name' and maybe other things
+// Assume branch name is last part of branch path
+if ( BRANCH_PATH ) {
```
Comment (niharikasingh) [inline lib/utils/env_vars.js:22]: I think this should be `process.env.BRANCH_PATH`, since the variable by itself doesn't exist yet.

**Review by niharikasingh — CHANGES_REQUESTED** [review]

File: lib/utils/env_vars.js (hunk 3 above)
Comment (plocket) [reply lib/utils/env_vars.js:22]: Gorramit. You're right. Good eye!

-

**Review by niharikasingh — APPROVED** [review]

**Pull request merged**